### ISSUE TITLE
Why not join dotnet foundation?

### ISSUE THREAD

**Issue by rocketRobin** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#646]
Maybe that will provide more help for this project or something else.

**Comment by caleblloyd** [comment]
I'm not familiar with the process but we could research it.

**Comment by ajcvickers** [comment]
@caleblloyd We should talk about this. Probably a good idea and we can likey help. /cc @divega

**Comment by divega** [comment]
I am following up with @jongalloway on this.

**Comment by caleblloyd** [comment]
I have filled out the "Submit a Project" form on the .NET Foundation website.

**Comment by rocketRobin** [comment]
Hi @divega ,is there any news for this thread?

**Comment by mguinness** [comment]
@jongalloway Can you help us with this effort?

**Comment by jongalloway** [comment]
Sorry for the slow communication here. We've moved to a public tracker on GitHub for tracking project applications. The form mentioned above was just to register interest, and after that we had a 3 page word doc to fill out. I don't think we have that application from you, if you previously sent it in it's the same as the GitHub issue template.

Can you open a new issue using the New Project Application here: https://github.com/dotnet-foundation/projects/issues/new/choose

We're going to be reviewing project applications this week, and I'd like to get this one in.

**Comment by mguinness** [comment]
@caleblloyd I assume as maintainer you have authority to transfer the project to the .NET Foundation?

**Comment by jongalloway** [comment]
From looking at the [contribution stats](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/graphs/contributors), I'd recommend that you have @caleblloyd @yukozh and @lauxjpn sign the contribution agreement.

**Comment by yukozh** [comment]
I've already talked with Julia Liuson some years ago, but she think our project is not large enough to join.

**Comment by jongalloway** [comment]
After filling out a [new project application](https://github.com/dotnet-foundation/projects/issues/new/choose), the board votes on whether the project should be accepted. After that, contribution agreement is sent out via DocuSign.

**Comment by caleblloyd** [comment]
> I assume as maintainer you have authority to transfer the project to the .NET Foundation?

Yes, @yukozh , @lauxjpn , and I should all have access to do that

---

I am supportive of filling out the application.  Originally we had hoped that joining the .NET foundation would help with visibility, governance, and attracting more contributors.  I think those would all still be helpful.

@lauxjpn since you have been the lead developer on the latest cycle of changes, what is your opinion?  If you agree, would you like to fill out the application?  I am also happy to fill out the application.

**Comment by lauxjpn** [comment]
@caleblloyd  I think it is a good idea and if this projects gets excepted might provide users/companies with additional confidence to choose this one over the oracle provider.

I will create a temporary branch for the application process on our repository and copy the template in there, so we can all work on it together until we are satisfied before submitting it.

It's probably a good idea to peek at the ones that got accepted in the past to get a better understanding about what the foundation (board) is looking for.

**Comment by jongalloway** [comment]
Let me know if you have any questions, and don't worry about getting everything right in the application. We have our monthly board meeting this Friday, so if you are able to get an application posted before Friday we can include for this month - otherwise we'd need to wait for next month.

**Comment by lauxjpn** [comment]
We are not going to make the board meeting for this month but will be ready for the one in December.

**Comment by mguinness** [comment]
Also referencing https://github.com/mysql-net/MySqlConnector/issues/535 in case @bgrainger has an interest.

**Comment by lauxjpn** [comment]
I will try to write some draft before the new year.

**Comment by yukozh** [comment]
happy new year.

**Comment by snithyanantham** [comment]
Any update on this

**Comment by lauxjpn** [comment]
No, there has been no progress from our side.

**Comment by JeremyLikness** [comment]
What's the action item here? #949 is still in draft, so does someone need to update that form and get it submitted, or did I miss something in the thread and it's already been sent over?

**Comment by lauxjpn** [comment]
> What's the action item here? #949 is still in draft, so does someone need to update that form and get it submitted, or did I miss something in the thread and it's already been sent over?

No, that is exactly what we still need to do:

> No, there has been no progress from our side.

**Comment by ElderJames** [comment]
I think this is a good suggestion.
I've actually contributed my own project [Ant Design Blazor](https://dotnetfoundation.org/projects/ant-design-blazor) to the foundation and completed all the processes, so I can offer some help.

**Comment by mguinness** [comment]
On an unrelated note but pertaining to Pomelo Foundation, should we apply for [Package ID prefix reservation](https://docs.microsoft.com/en-us/nuget/nuget-org/id-prefix-reservation) in NuGet?

**Comment by bgrainger** [comment]
I did this for "MySqlConnector" and it's a fairly simple process. (Though not sure if it'll be a problem if https://www.nuget.org/profiles/PomeloFoundation only owns 5 of the current `Pomelo.*` packages; you may have to consolidate ownership under one account first before getting a prefix reservation. I don't remember exactly how it works.)

**Comment by jcjiang** [comment]
Hi there! I'm a PM on the NuGet team and happy to help here. The consolidation isn't a strict requirement and I am happy to reserve for PomeloFoundation for now.

Since there is a number of community contributions, I assume a shared namespace would be best? That would give PomeloFoundation's Pomelo packages a verified status, but still allow contributors to publish within the prefix. Let me know what y'all think!

**Comment by lauxjpn** [comment]
@jcjiang Thanks for joining in!

I just [posted a couple of lists](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/issues/1172#issuecomment-979594568) in regards to packages with the "Pomelo." prefix.

Of the packages not published by us, only the ones from @ChaosEngine seem to be significant.
Maybe he is up to joining PomeloFoundation and to make this an official package, in which case we could close down the namespace to members of the PomeloFoundation as far as I am concerned.

The rest of the packages seem to be either very old forks, newer forks reviving the old and weirdly named [Pomelo.EntityFrameworkCore.Lolita](https://github.com/PomeloFoundation/Lolita) repo, which [should probably be renamed](https://github.com/PomeloFoundation/Lolita/issues/22), or packages doing something completely different with a very low download count.

The (nuget.org) authors of the other packages are (in order of significance):

| Authors |
| :--- |
| ChaosEngine |
| he1a2s0 |
| Mozbert |
| tcfialho |
| matocibo |
| tubo |

/cc @yukozh, @he1a2s0, @crozone

**Comment by ChaosEngine** [comment]
> Of the packages not published by us, only the ones from @ChaosEngine seem to be significant. Maybe he is up to joining PomeloFoundation and to make this an official package, in which case we could close down the namespace to members of the PomeloFoundation as far as I am concerned.
> 

Sure I'll try to conform, play along and do what is necessary for the greater good; being a team player etc. :-)
If some refactoring or namespace, package, assembly renaming is needed - fine, sign me in!

**Comment by yukozh** [comment]
[Lolita](https://github.com/PomeloFoundation/Lolita) is an important package. It was found due to EF core has a poor bulk update/delete SQL commands generation support. Due to EF Core 3.X has lots of breaking changes made the project hard to upgrade. But recently, I upgraded that to .NET 6.

**Comment by mguinness** [comment]
With the release of EF Core 7 we now have `ExecuteUpdate` and `ExecuteDelete` to help with bulk changes.

**Comment by lauxjpn** [comment]
> [Lolita](https://github.com/PomeloFoundation/Lolita) is an important package. It was found due to EF core has a poor bulk update/delete SQL commands generation support. Due to EF Core 3.X has lots of breaking changes made the project hard to upgrade. But recently, I upgraded that to .NET 6.

@yukozh If this package should not be deprecated, it [should be renamed](https://github.com/PomeloFoundation/Lolita/issues/22).

**Comment by crozone** [comment]
> With the release of EF Core 7 we now have ExecuteUpdate and ExecuteDelete to help with bulk changes.

For reference: https://learn.microsoft.com/en-us/ef/core/what-is-new/ef-core-7.0/whatsnew#executeupdate-and-executedelete-bulk-updates

This will obsolete Lolita for EF7 and up. It's still worth keeping it around (and renaming it) for EF Core 6 and below.
